NetKet/Operator/MatrixWrapper/direct_matrix_wrapper.hpp

```diff
@@ -47,7 +47,7 @@ class DirectMatrixWrapper : public AbstractMatrixWrapper<State> {
     State result(dim_);
     result.setZero();
 
-    for (size_t i = 0; i < dim_; ++i) {
+    for (int i = 0; i < dim_; ++i) {
```

Comment:
It's not particularly important, but why?

Reply:
it triggers a warning at compile time...

Reply:
Fine. But at some point it would be nice to convert all those Hilbert space dimensions to `size_t`. For now let's leave it like that.